six/modules/c++/six/include/six/NITFReadControl.h

```diff
@@ -222,6 +223,10 @@ class NITFReadControl : public ReadControl
 
         return (iCat == "LEG");
     }
+
+    // We need this for one of the load overloadings
+    // to prevent data from being deleted prematurely
+    mem::ScopedCopyablePtr<nitf::IOStreamReader> mStreamReader;
```

Comment:
So it's a tradeoff... this puts a weird special case as a member variable but in exchange doesn't change the public APIs for the `load(nitf::IOInterface& )` cases.  But what happens if we have another `CustomIO` or other overloading that also needs a smart pointer.  I think what would be cleaner is to have this simply hold a `mem::SharedPtr<nitf::IOInterface>` and always set it for all the overloadings.  That will change a couple load() overloadings, but I think it's worth it.  Thoughts?